### ISSUE TITLE
Fine-tuning your own dataset

### ISSUE THREAD

**Issue by kenken0911** [openvla/openvla#38]
Hello,

I am a student from Japan working on development using Denso industrial robots and OpenVLA. I would like to express my sincere gratitude for your remarkable research and development. I am eager to spread this wonderful technology in Japan. I believe that the integration of advanced AI technologies like OpenVLA with industrial robots will lead to significant growth and innovation in the robotics industry, opening up new possibilities for automation and intelligent manufacturing.

Currently, I am working on fine-tuning OpenVLA using a custom dataset. The specifications of the equipment I am using are as follows:

1. Industrial Robot:
   * Vertical multi-joint industrial robot: VP-6242
   * Pendant: TP-RC7/8

2. End Effector:
   * IAI electric gripper
   * Model: RCP2-GRST-I-20P-1-80-P3-S-A1
   * 20mm square pulse motor, reduction ratio 1/1 high-speed type, 80mm stroke

3. Camera:
   * Realsense D415

4. Inference Computer:
   * GPU: TUF-RTX4090-24G-GAMING
   * Memory: CMN64GX4M2Z3200C16 128GB
   * OS: Ubuntu Desktop 22.04 LTS

5. Development Environment:
   * GPU: NVIDIA RTX 3090
   * RAM: 64GB
   * CPU: Intel Core i9-11900K
   * OS: Ubuntu 20.04
   * Others: CUDA 11.2, PyTorch 1.10

**I would appreciate it if you could provide guidance on the following points:

1. The procedure for creating a custom dataset and the specifications for its format.
2. Detailed steps for fine-tuning the model using the dataset.**

I apologize for the inconvenience, but your assistance would be greatly appreciated. Thank you very much for your time and support.

Best regards
![Image (3)](https://github.com/user-attachments/assets/911704c8-202a-4787-b546-3b40f20d51cf)
![Image (2)](https://github.com/user-attachments/assets/900bb2f1-41b9-4bbb-a592-9603a3320ddf)
![Image (1)](https://github.com/user-attachments/assets/1e843d57-4713-48cf-8adf-164f845da119)

**Comment by kpertsch** [comment]
Hi! Thanks for reaching out! Your setup looks well-suited to try OpenVLA finetuning on it!
To get started, can you please take a closer look at the instructions we have in our readme: https://github.com/openvla/openvla?tab=readme-ov-file#fine-tuning-openvla-via-lora

They give a pretty detailed walkthrough of how to finetune on your own data, including how you convert your dataset into our data format, and an example script you can use for running the finetuning.

Your inference setup with a 4090 should be fast enough to inference the model at 6Hz!
For training, if the 3090 or 4090 is all the compute you have available you may need to play around with parameter efficient finetuning strategies like LoRA and gradient accumulation. If you have a slightly larger GPU so you can fit batches of 8-16 samples, finetuning should be easy.
